client/lib/i18n-utils/localize-path.js

```diff
@@ -0,0 +1,67 @@
+import { isMagnificentLocale, urlLocalizationMapping } from '@automattic/i18n-utils';
+import i18n from 'i18n-calypso';
+
+const DEFAULT_ORIGIN = 'https://wordpress.com';
+
+function getDefaultLocale() {
+	return i18n.getLocaleSlug?.() ?? 'en';
+}
+
+function suffixLocalizedPath( path, locale ) {
+	if ( ! isMagnificentLocale( locale ) ) {
+		return path;
+	}
+	return `${ path }${ locale }/`;
+}
+
+function normalizePath( urlLocalizationMappingFn ) {
+	return ( path, locale, isLoggedIn ) => {
+		const url = new URL( path, DEFAULT_ORIGIN );
+		const output = urlLocalizationMappingFn( url, locale, isLoggedIn );
+		return output.href.replace( DEFAULT_ORIGIN, '' );
+	};
+}
+
+const pathLocalizationMapping = {
+	'/theme/': normalizePath( urlLocalizationMapping[ 'wordpress.com/theme/' ] ),
+	'/themes/': normalizePath( urlLocalizationMapping[ 'wordpress.com/themes/' ] ),
+	'/log-in/': normalizePath( urlLocalizationMapping[ 'wordpress.com/log-in/' ] ),
+	'/new/': ( path, localeSlug, isLoggedIn ) => {
+		return isLoggedIn ? path : suffixLocalizedPath( path, localeSlug );
+	},
+	'/setup/': ( path, localeSlug, isLoggedIn ) => {
+		return isLoggedIn ? path : suffixLocalizedPath( path, localeSlug );
+	},
```

Comment:
Should this include `/start/`? (Don't know, just know there is /start and /new and both should be localized / logged-out-able)